nornir_nautobot/plugins/tasks/dispatcher/mikrotik_routeros.py

```diff
@@ -92,3 +92,58 @@ def get_config(  # pylint: disable=R0913,R0914
         cls._save_file(logger, backup_file, _running_config)
 
         return Result(host=task.host, result={"config": _running_config})
+
+    @staticmethod
+    def merge_config(
+        task: Task,
+        logger,
+        obj,
+        config: str,
+    ) -> Result:
+        """Send configuration to merge on the device.
+
+        Args:
+            task (Task): Nornir Task.
+            logger (NornirLogger): Custom NornirLogger object to reflect job_results (via Nautobot Jobs) and Python logger.
+            obj (Device): A Nautobot Device Django ORM object instance.
+            config (str): The config set.
+
+        Raises:
+            NornirNautobotException: Authentication error.
+            NornirNautobotException: Timeout error.
+            NornirNautobotException: Other exception.
+
+        Returns:
+            Result: Nornir Result object with a dict as a result containing what changed and the result of the push.
+        """
+        NETMIKO_FAIL_MSG = ["bad", "failed", "failure"]  # pylint: disable=C0103
+        logger.log_success(obj, "Config merge starting")
+
+        try:
+            config_list = config.splitlines()
+            push_result = task.run(
+                task=netmiko_send_config,
+                config_commands=config_list,
+            )
+        except NornirSubTaskError as exc:
+            logger.log_failure(obj, f"Failed with error: `{exc.result.exception}`")
+            raise NornirNautobotException() from exc
+
+        if any(msg in push_result[0].result.lower() for msg in NETMIKO_FAIL_MSG):
```

Comment:
can't this be done in deterministic way without relaying on text? status code is not available?

Reply:
seems the "changed" status for nornir_netmiko tasks is always ==False. Otherwise the status is always succeed even for failed tasks.